wrappers/adaptive/popcorn.HTMLAdaptiveMediaElement.js

```diff
@@ -192,26 +229,67 @@
                       media.src = fallbackMedia;
                     }
                   });
-                  player.on(dashjs.MediaPlayer.events.SOURCE_INITIALIZED, function() {
+                  player.on(dashjs.MediaPlayer.events.SOURCE_INITIALIZED, function () {
                     player.setTrackSwitchModeFor('video', 'alwaysReplace');
                     player.setTrackSwitchModeFor('audio', 'alwaysReplace');
                     player.setAutoSwitchQualityFor('video', true);
                     player.setAutoSwitchQualityFor('audio', true);
-                    player.setInitialBitrateFor('video', 99999999);
                     player.setInitialBitrateFor('audio', 99999999);
                   });
+                  player.on(dashjs.MediaPlayer.events.STREAM_INITIALIZED, function () {
+                    var bitrates = player.getBitrateInfoListFor('video');
+                    if (bitrates && bitrates.length) {
+                      bitrates = bitrates.map(function (q, idx) {
+                        q.resolution = q.height;
+                        q.value = idx;
+                        return q;
+                      });
+                      bitrates.push({ resolution: "auto", value: "auto" });
+                      media.qualities = bitrates;
+                    } else {
+                      media.qualities = [];
+                    }
+                    media.quality = player.getQualityFor('video');
+                    media.dispatchEvent("loadedbitrate");
+                    updateQuality = function (quality) {
+                      if (quality === "auto") {
```

Comment:
string literal is a way to make a typo, may be make constant for this literal, or use something like -1 as literal when auto. in addition to that http://cdn.dashjs.org/latest/jsdoc/module-MediaPlayer.html#setQualityFor, argument of setQualityFor is number, and you can have in quality string (auto) or number. Mixing type of variable is not seeming too reliable.

Reply:
and the similar I see for HLS below.

Reply:
done